File: ts/examples/voiceAgent.ts
import 'dotenv/config';
import { type JobContext, ServerOptions, cli, defineAgent, voice } from '@livekit/agents';
import * as openai from '@livekit/agents-plugin-openai';
import * as reson8 from '@reson8-labs/agents-plugin-reson8';
import { fileURLToPath } from 'node:url';

export default defineAgent({
  entry: async (ctx: JobContext) => {
    await ctx.connect();

    const session = new voice.AgentSession({
      stt: new reson8.STT({
        eagerTurnProbability: 0.5,
        finalTurnProbability: 0.7,
      }),
      llm: new openai.LLM(),
      tts: new openai.TTS(),
      turnHandling: {
        turnDetection: 'stt',
        preemptiveGeneration: { enabled: true },
      },
    });

    const agent = new voice.Agent({
      instructions: 'You are a helpful voice assistant.',
    });

    await session.start({ agent, room: ctx.room });
    await session.say('Hallo, hoe kan ik je helpen?');
  },
});

cli.runApp(new ServerOptions({ agent: fileURLToPath(import.meta.url) }));

File: python/examples/voice_agent.py
from dotenv import load_dotenv
from livekit.agents import AutoSubscribe, JobContext, WorkerOptions, cli
from livekit.agents.voice import VoiceAgent
from livekit.plugins import openai, reson8

load_dotenv()


async def entrypoint(ctx: JobContext):
    await ctx.connect(auto_subscribe=AutoSubscribe.AUDIO_ONLY)

    # reson8.STT streams with server-side turn detection, which keeps
    # voice-agent responses snappy. Any language is supported: omit `language`
    # to auto-detect, or pass any code (e.g. language="nl") to pin it.
    agent = VoiceAgent(
        stt=reson8.STT(),
        llm=openai.LLM(),
        tts=openai.TTS(),
    )
    agent.start(ctx.room)

    await agent.say("Hallo, hoe kan ik je helpen?")


if __name__ == "__main__":
    cli.run_app(WorkerOptions(entrypoint_fnc=entrypoint))
